src/main/scala/fi/oph/koski/valpas/valpasrepository/ValpasKuntailmoitusRepository.scala

```diff
@@ -144,5 +144,17 @@ class ValpasKuntailmoitusRepository(
     )
   }
 
+  def queryByKunta(kuntaOid: Organisaatio.Oid): Either[HttpStatus, Seq[ValpasKuntailmoitusLaajatTiedotJaOppijaOid]] = {
+    HttpStatus.foldEithers(
+      runDbSync(
+        Ilmoitukset
+          .filter(_.kuntaOid === kuntaOid)
+          .join(IlmoitusLisätiedot).on(_.uuid === _.ilmoitusUuid) // TODO: Tämän pitäisi olla left join ja queryn toimia, vaikka koko lisätiedot taulu olisi tyhjä: lisätiedot ovat oppivelvollisuusrekisterin ulkopuolista dataa
```

Comment:
ks `joinLeft`

Reply:
Tästä saattaa tulla isompikin refaktorointi, koska näyttäisi että ilman lisätietoja ei datoja saa mäpättyä `ValpasKuntailmoitusLaajatTiedot` -luokkaan. En osaa sanoa kuinka laajalle se vaikuttaa, mutta jonkin verran hommaa siitä varmasti tulee. Pitää jutella vielä, mikä on se tilanne ettei lisätietoja kannasta löydy.

Reply:
Tein tästä oman lapun

Reply:
käsittääkseni tällainen tilanne voi syntyä, jos joku oppija esim. pyytää poistamaan kaikki ei-oleelliset henkilötiedot rekisteristä. Silloin kaikki lisätiedot pitää poistaa. Voidaan toki toteuttaa se niinkin, että poistetaan vaan JSONin sisältämä data, mutta jotenkin koko rivin poisto kuulostaa helpommalta toteuttaa, kun kuitenkin manuaalioperaatioina näitä joudutaan tekemään.